nemo/collections/asr/jasper.py

```diff
@@ -317,3 +317,179 @@ def forward(self, encoder_output):
             return logits
 
         return F.softmax(logits, dim=-1)
+
+
+class JasperDecoderForSpkrClass(TrainableNM):
+    """
+    Jasper Decoder creates the final layer in Jasper that maps from the outputs
+    of Jasper Encoder to the embedding layer followed by speaker based softmax loss.
+
+    Args:
+        feat_in (int): Number of channels being input to this module
+        num_classes (int): Number of unique speakers in dataset
+        emb_sizes (list) : shapes of intermidiate embedding layers (we consider speaker embbeddings from 1st of this layers)
+        gram (bool) : If we should consider gram based pooling
+        superVector (bool) : If we should unify  x-vector based pooling with gram based pooling
+        init_mode (str): Describes how neural network parameters are
+            initialized. Options are ['xavier_uniform', 'xavier_normal',
+            'kaiming_uniform','kaiming_normal'].
+            Defaults to "xavier_uniform".
```

Comment:
Can you add defaults to all args in your docstring?

Reply:
Can you remove the gram and super_vector args and just add a mode option with xvector, gram, and super as options?>

Reply:
Clean thought. Yes, added pool_mode instead of giving too many options for user 👍🏻